### DISCUSSION TITLE
Unpaginated Permissions Endpoint called in Management Plugin UI

### DISCUSSION THREAD

**Discussion by raynigon** [rabbitmq/rabbitmq-server#8452] [Other]
### Describe the bug

The Management Plugin UI calls the permission endpoint unpaginated. This leads to OOM errors if a few thousand permissions were created.

### Reproduction steps

1. Create a few thousand permissions (~50k)
2. Open the management UI
3. Go to the admin settings
4. list users and permissions
5. Server restarts and UI is left in broken state

### Expected behavior

The page loads as expected and the permissions list is paginated.

### Additional context

Since this can not be solved by scaling the instance count, it would be nice to paginate the endpoint as it is done with the users endpoint. The permissions response with 50k permissions is about 10MB.

**Comment by michaelklishin** [comment]
In my 13 years as a user and contributor, I have never see someone else run a node out of memory because of 50K permissions.

You are welcome to contribute a PR.

**Reply by michaelklishin** [reply]
One scenario where I can imagine this being a real scenario is one user per device. This can work up to a point with the internal database authN-authZ backend.

Sinner non-privileged users cannot grant themselves permissions, no RabbitMQaaS environments run into this, unlike with most other entities (channels or queues, for example).
